src/parameterizations/vertical/MOM_set_viscosity.F90

```diff
@@ -820,41 +821,90 @@ subroutine set_viscous_BBL(u, v, h, tv, visc, G, GV, CS)
 
 end subroutine set_viscous_BBL
 
-function set_v_at_u(v, h, G, i, j, k)
+function set_v_at_u(v, h, G, i, j, k, OBC)
   type(ocean_grid_type),                     intent(in) :: G
   real, dimension(SZI_(G),SZJB_(G),SZK_(G)), intent(in) :: v
   real, dimension(SZI_(G),SZJ_(G),SZK_(G)),  intent(in) :: h
   integer,                                   intent(in) :: i, j, k
+  type(ocean_OBC_type), pointer, optional               :: OBC
   real                                                  :: set_v_at_u
+  integer                                            :: n
   ! This subroutine finds a thickness-weighted value of v at the u-points.
   real :: hwt(4)           ! Masked weights used to average v onto u, in H.
   real :: hwt_tot          ! The sum of the masked thicknesses, in H.
+  type(OBC_segment_type), pointer :: segment
 
   hwt(1) = (h(i,j-1,k) + h(i,j,k)) * G%mask2dCv(i,J-1)
   hwt(2) = (h(i+1,j-1,k) + h(i+1,j,k)) * G%mask2dCv(i+1,J-1)
   hwt(3) = (h(i,j,k) + h(i,j+1,k)) * G%mask2dCv(i,J)
   hwt(4) = (h(i+1,j,k) + h(i+1,j+1,k)) * G%mask2dCv(i+1,J)
+
+
+  if (present(OBC)) then
```

Comment:
A better solution (thanks @adcroft ) is to create two 2-d masks for this calculation (one each for interpolation to u and v points ) upon initialization , store in the visc control structure, and pass to set_v_at_u to be used in place of G%mask2dCv at no extra cost.